packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -323,4 +350,18 @@ class _DefaultTabControllerState extends State<DefaultTabController> with Single
       child: widget.child,
     );
   }
+
+  @override
+  void didUpdateWidget(DefaultTabController oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.length != widget.length) {
+      // If the length is shortened while the last tab is selected, we should
+      // automatically update the index of the controller to be the new last tab.
+      int newIndex;
+      if (_controller.index >= widget.length) {
+        newIndex = widget.length - 1;
```

Comment:
widget.length can be 0, is it safe to create a controller with index  = -1? Might be safer to use max(0, widget.length - 1)

It looks like DefaultTabController's constructor isn't checking its parameters. We should assert that length >=0; initialIndex >=0 && initialIndex < length

Reply:
Done, added `max(0, widget.length - 1)` here and `assert(length >= 0),
       assert(initialIndex >= 0 && initialIndex < length)` to `DefaultTabController`.